engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -405,6 +405,9 @@ def scc_deactivate_product
               # it is OK to remove it from SCC
               response = SccProxy.deactivate_product_scc(auth, @product, @system.system_token)
               handle_response(response)
+              # if the system does not have more non free products activated on SCC
+              # system should turn to hybrid
+              @system.payg! if no_more_activations scc_systems_activations
```

Comment:
Why is that `no_more_activations'`method needed, can't we just use the existing `find_hybrid_product_on_scc(request.headers).empty?`?

Reply:
Because that would require a new request to SCC, the information to check is available without the request

The product successfully removed should be skipped during the check of that info, that's the difference with calling SCC again